frontend/src/metabase/visualizations/visualizations/SmartScalar/SettingsComponents/PeriodsAgoOptionComponent.tsx

```diff
@@ -0,0 +1,158 @@
+import { useCallback, useMemo, useRef, useState } from "react";
+import type {
+  Dispatch,
+  KeyboardEvent,
+  MouseEvent,
+  SetStateAction,
+} from "react";
+import { Group, Text, Box } from "metabase/ui";
+import type { SelectedComparisonPeriodsAgo } from "metabase-types/api";
+import { NumberInputStyled } from "./PeriodsAgoOptionComponent.styled";
+import { MenuItemStyled } from "./MenuItem.styled";
+
+type PeriodsAgoOptionComponentProps = {
+  isSelected: boolean;
+  maxValue: number;
+  name: string;
+  onChange: (setting: { type: string; value?: number }) => void;
+  selectedValue?: SelectedComparisonPeriodsAgo;
+  setOpen: Dispatch<SetStateAction<boolean>>;
+  type: string;
+};
+
+export function PeriodsAgoOptionComponent({
+  isSelected,
+  maxValue,
+  name,
+  onChange,
+  selectedValue,
+  setOpen,
+  type,
+}: PeriodsAgoOptionComponentProps) {
+  const value = useMemo(() => {
+    if (!selectedValue) {
+      return null;
+    }
+
+    if (Number.isInteger(selectedValue.value)) {
+      return selectedValue.value;
+    }
+
+    return null;
+  }, [selectedValue]);
+  const minValue = 2;
+
+  const [inputValue, setInputValue] = useState(value ?? minValue);
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  // used to prevent accidental button click when mouseDown inside input field
+  // but dragged to highlight all text and accidentally mouseUp on the button
+  // outside the input field
+  const mouseDownInChildRef = useRef(false);
+  const handleChildMouseDownAndUp = useCallback(
+    (e: MouseEvent<HTMLInputElement>) => {
+      e.stopPropagation();
+      e.preventDefault();
+      inputRef.current?.select();
+
+      mouseDownInChildRef.current = true;
+    },
+    [],
+  );
```

Comment:
@JesseSDevaney I've noticed Mantine Menu has two [props](https://v6.mantine.dev/core/menu/?t=props) that might help us avoid these hacks:

* `Menu` has a `closeOnItemClick` prop
* `MenuItem` has a `closeMenuOnClick` prop

Can we look into using one of them? FWIW, I've used `MenuItem's` `closeMenuOnClick` in #36951